lib/eco/treelexer/lexer.py

```diff
@@ -80,24 +89,59 @@ def match_one(self, pattern):
         self.exactmatch = False
         return False
 
-    def match_star(self, pattern):
-        while self._inner_match(pattern.c):
-             pass
-        return True
+    def match_star(self, pattern, l=None):
+        if l is None or len(l) == 1:
+            while self._inner_match(pattern.c):
+                pass
+            return True
 
-    def match_plus(self, pattern):
+        if pattern.ng: # non-greedy
+            while True:
+                tmp = self.save_state()
+                if self.match_list(l[1:]):
+                    return True
+                self.load_state(tmp)
+                if self._inner_match(pattern.c):
+                    continue
+                return False
+        else: # greedy
+            stack = []
+            while True:
+                stack.append(self.save_state())
+                if self._inner_match(pattern.c):
+                    continue
+                break
+            max_la = self.la
+            if len(l) > 1:
+                while stack:
+                    self.load_state(stack.pop())
+                    if self.match_list(l[1:]):
+                        return True
+                    max_la = max(self.la, max_la)
+                self.la = max_la
+            return False
+
+    def match_plus(self, pattern, l=None):
         if not self._inner_match(pattern.c):
             # we have to at least match one
             return False
-        return self.match_star(pattern)
+        if l:
+            # convert RE_PLUS to RE_STAR
```

Comment:
Why?

Reply:
This is no longer needed and I removed it in the latest commit.